### ISSUE TITLE
[Bugsnag issue] - Item to delete is undefined in entityQueue

### ISSUE THREAD

**Issue by vinayvenu** [avniproject/avni-client#843]
We are seeing an unusually high number of these errors since release 3.37.6. See details in the link below

Plausible reason - We are running multiple syncs together. 

https://app.bugsnag.com/samanvay-research-and-development-foundation/avni-client/errors/63bc3395a8e958000777aa4a?filters[event.since]=30d&filters[error.status]=open&filters[app.release_stage][]=prod&filters[app.release_stage][]=production&filters[app.release_stage][]=uat&filters[device.manufacturer][type]=ne&filters[device.manufacturer][value]=Genymotion&pivot_tab=event

-- Figure out a way to disable foreground sync when background sync is on
-- Try not running the background job immediately when the app starts

**Comment by 1t5j0y** [comment]
Putting on hold for now since this is tied to a larger piece of handling multiple syncs in parallel.
